components/sup/src/manager.rs

```diff
@@ -1337,18 +1338,14 @@ impl Manager {
                                            stop_it)
     }
 
-    fn unload(&mut self,
-              ident: &PackageIdent,
-              shutdown_input: &ShutdownInput,
-              runtime: &mut Runtime) {
+    fn try_remove_spec_file(&self, ident: &PackageIdent) {
```

Comment:
Yeah, I was debating what to do with these methods (the other being `try_stop_service`). I agree using `try_` is probably not appropriate.

My goal was to indicate that these methods can fail, but instead of returning `Result` they handle the errors internally by logging appropriately. I did not want somebody to see `remove_spec_file` and think that it guaranteed the file was removed. I would like some way to distinguish between methods that propagate errors, and higher-level methods that can still fail but handle the failure case probably by logging. Maybe this distinction is not needed? What do people think? Other possible names would be `attempt_remove_spec_file` or `remove_spec_file_with_log`, but maybe these are even more confusing?

Reply:
I think I'd be most in favor of keeping the name (it is a good indicator of fallibility), and returning a `Result`. We can still log in this function (or move that to the callers, if our desire to log is context dependent). It's easy enough for callers to ignore the result with `.ok();` if they don't care about failure. I think we should get used to that idiom as an indicator that a function may not have succeeded, but that the caller doesn't behave differently.

Reply:
To add on to @baumanj's suggestion (which I agree with 100%)... we've come across a lot of places in the past where the fact that we swallow errors internally causes problems. If we were to get used to propagating errors, that would give us a bit more flexibility in how we handle things.

This also will require some new error types, I imagine. Coupled with some of the refinements detailed in https://github.com/habitat-sh/habitat/issues/6559, I could envision lots of rich error types... for errors that we choose to ignore, we could even have a nice helper function to chain onto calls that basically takes all that rich error information and logs it. Then it's really clear in the code which error cases we care about and which ones we don't, but we never lose that information, and if we do decide we need to act on it, we can easily do so.

Reply:
What about a `ResultExt` trait or something that would allow us to "lognore" errors. Instead of `try_remove_spec_file(…).ok()`, it could be something like `.ok_log_err()`?

Reply:
@baumanj Yup, that's exactly what I was thinking.

Reply:
Incidentally, I :heart: "lognore" :joy:

Reply:
While I completely agree this sounds like the correct way (`ok_log_err` is a great idea), I think it makes sense to hold off on implementing it in this PR. I will link #6559 to this discussion. I am happy to tackle #6559 if we feel it is a blocker for this to be merge.